### ISSUE TITLE
How to get anonymous module?

### ISSUE THREAD

**Issue by Thaina** [angular/angular.js#14733]
Since angular could attach `ng-app` attribute without value. And one document can exist only one element with `ng-app`. Why there are no api such as `angular.rootModule([])` or something to initial that module? Why we need to specify module name because the name is just that document

**Comment by gkalpak** [comment]
This concept of "root module" does not exist. The fact that you can only have one `ngApp` per document, does not mean that you can have only one app.

`ngApp` is just syntactic sugar for manually calling `angular.bootstrap` and passing in the DOM element on which you want to bootstrap an app and the modules you want to load. If you don't specify a module name in `ngApp`, then no (extra) module is loaded during bootstrap. I.e. no "root module" will be created for you.
(Note, that Angular will always prepend its core module (`ng`) when bootstrapping.)

So, basically, when you have `<some-element ng-app>` (without specifying a module), only the core `ng` module will be loaded. It is not considered a good practice, but you could add more stuff to it if you wanted to.
